subprojects/core/src/main/java/org/gradle/api/internal/file/DefaultBuildLayout.java

```diff
@@ -16,16 +16,17 @@
 
 package org.gradle.api.internal.file;
 
+import org.gradle.api.file.BuildLayout;
 import org.gradle.api.file.Directory;
-import org.gradle.api.file.SettingsLayout;
-import org.gradle.api.internal.SettingsInternal;
+import org.gradle.api.initialization.Settings;
 
 
-public class DefaultSettingsLayout extends AbstractFileSystemLayout implements SettingsLayout {
-    private final SettingsInternal settings;
+public class DefaultBuildLayout implements BuildLayout {
+    protected final FileFactory fileFactory;
+    private final Settings settings;
```

Comment:
❌ This type should not need a `Settings` instance. Its root directory should be the root directory of the build, taken from the `BuildLocations` rather than the mutable project directory of the root project.

This can be fixed in a follow up PR.

It looks like `Settings.getRootDir()` is doing the wrong thing, but I would just leave that unchanged and we can deprecate it later.

Reply:
Added https://github.com/gradle/gradle/issues/26521 to handle that bit later.